### PR TITLE
API Reference v1.22

### PR DESCRIPTION
**Pull request by feloy**
1. Move the assets necessary to build the API Reference from the api-ref-generator submodule (https://github.com/kubernetes-sigs/reference-docs) to this repository. This is in preparation to add translation assets for the API Reference (https://github.com/kubernetes/website/pull/27842)

 2. Generated files for the API Reference for v1.22 (beta.0 at this time)

### CODE REVIEW

**Comment by kbhawkey** [comment]
👀

**Comment by sftim** [comment]
/milestone 1.22
/hold

This PR should target the _dev-1.22_ branch, right?

**Comment by tengqm** [comment]
It is still too early to propose this ... changes are still coming in. For example: https://github.com/kubernetes/kubernetes/commit/29b3fa782631ab65c7e1f48fecef58c5365577c8#diff-ce7b64e72add3d44567b11ab1a4b1e6bfc9345a86b4a352283140e4974776c4d

File: api-ref-assets/config/toc.yaml [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    key: io.k8s.apimachinery.pkg.apis.meta.v1.Status
+  - name: TypedLocalObjectReference
+    key: io.k8s.api.core.v1.TypedLocalObjectReference
+skippedResources:
+- APIGroup
+- APIGroupList
+- APIResourceList
+- APIVersions
+- Eviction
+- Scale
```
Comment (sftim) [inline api-ref-assets/config/toc.yaml:264]: Not a v1.22 thing but it does make me wonder: why do we skip eviction and scale subresources?

**Comment by PI-Victor** [comment]
whatever happened to this PR, i'm guessing we're not gonna merge it for 1.22?

**Comment by sftim** [comment]
We do need an API reference update (two of them, in fact, unless we're planning to drop the one-page reference already?)

**Comment by sftim** [comment]
Merging this might be better than not (provided we address the conflict in `README.md`). Even better, though, is to do an up to date generation based on the code frozen for release.

**Comment by feloy** [comment]
> Merging this might be better than not (provided we address the conflict in `README.md`). Even better, though, is to do an up to date generation based on the code frozen for release.

I just rebased this PR. The PR includes the latest swagger.json file, which is I can imagine the frozen one.

**Comment by feloy** [comment]
@sftim should this PR be based on main, or dev-1.22?

**Comment by sftim** [comment]
It'll be more useful to base this on _dev-1.22_ @feloy - @PI-Victor I hope you agree?

I think we could cherry-pick the commits anyway but it's easier if the merge base starts off as the one we want.

**Comment by PI-Victor** [comment]
yes, i'd say the same, i wanna merge this into dev-1.22 and then merge dev-1.22 into main after the rest of the stuff is generated as well

**Comment by feloy** [comment]
I won't be available now for the retarget, I'll do it asap today

**Comment by feloy** [comment]
I've created a new PR targeting dev-1.22: https://github.com/kubernetes/website/pull/29217

**Pull request closed without merging**